backend/zkppSalt.js

```diff
@@ -182,18 +182,13 @@ const verifyChallenge = (contractID: string, r: string, s: string, userSig: stri
   return sig.byteLength === macBuf.byteLength && timingSafeEqual(sig, macBuf)
 }
 
-export const registrationKey = async (contractID: string, b: string): Promise<false | {s: string; p: string; sig: string;}> => {
-  const record = await getZkppSaltRecord(contractID)
```

Comment:
Just curious, why isn't this needed any longer? I wouldn't expect a "validation improvements" PR to modify this type of logic

Reply:
You're right. This mostly is validation improvements, but I made some other related small improvements that I'd forgotten in my previous PR.

The change being made here is because `registrationKey` doesn't need to be async after the latest changs to salt registration, because the salt is registered after redeeming a token, and a provisional ID (the username, which is known at the time this is called) is used instead of the real ID (the contract ID, which is only known later).

Reply:
I didn't address this as I don't think there's anything to address, but to expand on my previous comment. Note how `contractID` was changed to `provisionalId`. This change, which is just a name change, doesn't affect functionality but better reflects usage, since, as mentioned earlier, a `contractID` isn't used (when `registationKey` is called) as it's unknown.